browser/infobars/brave_ipfs_always_start_infobar_delegate.cc

```diff
@@ -0,0 +1,109 @@
+
+#include "base/functional/callback_helpers.h"
+#include "brave/components/ipfs/ipfs_service.h"
+#include "brave/components/ipfs/pref_names.h"
+#include "brave/components/l10n/common/localization_util.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "components/prefs/pref_service.h"
+
+BraveIPFSAlwaysStartInfoBarDelegateFactory::
+    BraveIPFSAlwaysStartInfoBarDelegateFactory(ipfs::IpfsService* ipfs_service,
+                                               PrefService* local_state)
+    : local_state_(local_state), ipfs_service_(ipfs_service) {}
+
+std::unique_ptr<BraveConfirmInfoBarDelegate>
+BraveIPFSAlwaysStartInfoBarDelegateFactory::Create() {
+  if (!local_state_ || local_state_->GetBoolean(kIPFSAlwaysStartMode) ||
+      (!disable_resolve_method_check_for_testing_ &&
+       local_state_->GetInteger(kIPFSResolveMethod) !=
+           static_cast<int>(ipfs::IPFSResolveMethodTypes::IPFS_LOCAL))) {
+    return nullptr;
+  }
+
+  if (local_state_->GetBoolean(kIPFSAlwaysStartInfobarShown)) {
+    return nullptr;
+  }
+
+  return base::WrapUnique(
+      new BraveIPFSAlwaysStartInfoBarDelegate(ipfs_service_, local_state_));
+}
+
+infobars::InfoBarDelegate::InfoBarIdentifier
+BraveIPFSAlwaysStartInfoBarDelegateFactory::GetInfoBarIdentifier() const {
+  return BraveConfirmInfoBarDelegate::BRAVE_IPFS_ALWAYS_START_INFOBAR_DELEGATE;
+}
+
+void BraveIPFSAlwaysStartInfoBarDelegateFactory::
+    DisableResolveMethodCheckForTesting() {
+  disable_resolve_method_check_for_testing_ = true;
+}
+
+BraveIPFSAlwaysStartInfoBarDelegate::BraveIPFSAlwaysStartInfoBarDelegate(
+    ipfs::IpfsService* ipfs_service,
+    PrefService* local_state)
+    : local_state_(local_state), ipfs_service_(ipfs_service) {}
+
+BraveIPFSAlwaysStartInfoBarDelegate::~BraveIPFSAlwaysStartInfoBarDelegate() =
+    default;
+
+// ConfirmInfoBarDelegate
+infobars::InfoBarDelegate::InfoBarIdentifier
+BraveIPFSAlwaysStartInfoBarDelegate::GetIdentifier() const {
+  return BRAVE_IPFS_ALWAYS_START_INFOBAR_DELEGATE;
+}
+
+std::u16string BraveIPFSAlwaysStartInfoBarDelegate::GetMessageText() const {
+  return brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_BRAVE_IPFS_ALWAYS_START_INFOBAR_TEXT);
+}
+
+int BraveIPFSAlwaysStartInfoBarDelegate::GetButtons() const {
+  return BUTTON_OK | BUTTON_CANCEL;
+}
+
+std::vector<int> BraveIPFSAlwaysStartInfoBarDelegate::GetButtonsOrder() const {
+  return {BUTTON_OK, BUTTON_CANCEL};
+}
+
+std::u16string BraveIPFSAlwaysStartInfoBarDelegate::GetButtonLabel(
+    InfoBarButton button) const {
+  if (button == BUTTON_CANCEL) {
+    return brave_l10n::GetLocalizedResourceUTF16String(
+        IDS_BRAVE_IPFS_ALWAYS_START_INFOBAR_NO);
+  }
+  return brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_BRAVE_IPFS_ALWAYS_START_INFOBAR_OK);
+}
+
+bool BraveIPFSAlwaysStartInfoBarDelegate::Accept() {
+  if (local_state_) {
+    local_state_->SetBoolean(kIPFSAlwaysStartMode, true);
```

Comment:
I think you can just subscribe in the IpfsService on this pref and start daemon then. And you won't need to launch service from settings\infobar in this case.

Reply:
done